relayer/rly/cosmos_relayer.go

```diff
@@ -138,21 +138,46 @@ func (relayer *CosmosRelayer) GetChannels(ctx context.Context, chainID string) (
 }
 
 // Implements Relayer interface
-func (relayer *CosmosRelayer) StartRelayer(ctx context.Context, pathName string) error {
-	return relayer.CreateNodeContainer(pathName)
+func (relayer *CosmosRelayer) StartRelayer(ctx context.Context, rep ibc.RelayerExecReporter, pathName string) error {
+	return relayer.createNodeContainer(pathName)
 }
 
 // Implements Relayer interface
-func (relayer *CosmosRelayer) StopRelayer(ctx context.Context) error {
-	if err := relayer.StopContainer(); err != nil {
+func (relayer *CosmosRelayer) StopRelayer(ctx context.Context, rep ibc.RelayerExecReporter) error {
+	if err := relayer.stopContainer(); err != nil {
 		return err
 	}
-	stdout := new(bytes.Buffer)
-	stderr := new(bytes.Buffer)
-	_ = relayer.pool.Client.Logs(docker.LogsOptions{Context: ctx, Container: relayer.container.ID, OutputStream: stdout, ErrorStream: stderr, Stdout: true, Stderr: true, Tail: "50", Follow: false, Timestamps: false})
+	finishedAt := time.Now()
+
+	stdoutBuf := new(bytes.Buffer)
+	stderrBuf := new(bytes.Buffer)
+	_ = relayer.pool.Client.Logs(docker.LogsOptions{
+		Context:      ctx,
+		Container:    relayer.container.ID,
+		OutputStream: stdoutBuf,
+		ErrorStream:  stderrBuf,
+		Stdout:       true,
+		Stderr:       true,
+		Tail:         "50",
+		Follow:       false,
+		Timestamps:   false,
+	})
+
+	stdout := stdoutBuf.String()
+	stderr := stderrBuf.String()
+
+	rep.TrackRelayerExec(
+		relayer.container.Name,
+		relayer.container.Args, // Is this correct for the command?
+		stdout, stderr,
+		relayer.container.State.ExitCode, // Is this accurate?
```

Comment:
Re your comments: Good questions. 

Tough to know for sure without running the suite and observing. I personally don't know this docker pkg well enough to say one way or the other.